### ISSUE TITLE
@auth transform applies each rule to all the operations, not just the ones specified

### ISSUE THREAD

**Issue by ceich** [aws-amplify/amplify-cli#93]
**Do you want to request a feature or report a bug?**
Bug

**What is the current behavior?**

The sample schema added for "Objects with fine-grained access control" (offered only when using User Pool authorization type) has the following annotation for the Task type:
```
  @auth(rules: [
      {allow: groups, groups: ["Managers"], mutations: [create, update, delete]},
      {allow: groups, groups: ["Employees"], queries: [get, list]}
    ])
```
I interpret this to mean that only Managers can invoke the mutations, while only Employees can invoke the queries (with the practical assumption that any user that is a Manager is also an Employee).

**Steps to reproduce the issue:**

1. Add a GraphQL API with User Pool authorization
2. Select the "Objects with fine-grained access control" project type
3. Examine the generated `*Task.*` VTL files.
4. The mutation _request_ templates require that the invoker be in _both_ Managers and Employees groups.
5. The queries _response_ templates require that the invoker be in _either_ Managers or Employees.

**What is the expected behavior?**

1.  The three mutations should require that the invoker be a member of Managers.
2. The two queries should require that the invoker be a member of Employees.

**Additional environment details**

* **OS version**:  macOS 10.13.6
* **Output of ``amplify --version``**: 0.1.15
* **Did this work in previous versions?**: unknown

**Comment by mikeparisstuff** [comment]
I'll take a look at this.

**Comment by mikeparisstuff** [comment]
Just fixed in the above PR. Thanks for surfacing this bug.

**Comment by kaustavghosh06** [comment]
Fixed pushed with latest npm publish -> 0.1.18
